src/explorer/siteNodeBase.ts

```diff
@@ -165,37 +165,62 @@ export class SiteNodeBase extends NodeBase {
+            await this.webSiteClient.webApps.listPublishingCredentialsSlot(this.site.resourceGroup, siteName, util.extractDeploymentSlotName(this.site));
+        const config = !isSlot ?
+            await this.webSiteClient.webApps.getConfiguration(this.site.resourceGroup, siteName) :
+            await this.webSiteClient.webApps.getConfigurationSlot(this.site.resourceGroup, siteName, util.extractDeploymentSlotName(this.site));
+        const oldDeployment = !isSlot ?
+            await this.webSiteClient.webApps.listDeployments(this.site.resourceGroup, siteName) :
+            await this.webSiteClient.webApps.listDeploymentsSlot(this.site.resourceGroup, siteName, util.extractDeploymentSlotName(this.site));
 
         if (config.scmType !== 'LocalGit') {
-            let input = await window.showErrorMessage(`Local Git Deployment is not set up. Set it up in the Azure Portal.`, `Go to Portal`)
-            if (input === 'Go to Portal') {
-                this.openInPortal();
+            let input;
+            let oldScmType = config.scmType;
+            let updateConfig = config;
+            updateConfig.scmType = 'LocalGit';
+
+            if (oldScmType !== 'None') {
+                input = await window.showWarningMessage(`Deployment source for "${siteName}" is set as "${oldScmType}".  Change to "LocalGit"?`, 'Yes');
+            }
+
+            if (oldScmType === 'None' || input === 'Yes') {
+                !isSlot ?
+                    await this.webSiteClient.webApps.updateConfiguration(this.site.resourceGroup, siteName, updateConfig) :
+                    await this.webSiteClient.webApps.updateConfigurationSlot(this.site.resourceGroup, siteName, updateConfig, util.extractDeploymentSlotName(this.site));
+                console.log('Updated Config!');
+            } else {
+                throw new UserCancelledError();
             }
-            throw new Error(`Local Git Deployment is not set up. Set it up in the Azure Portal.`);
         }
 
         const username = publishCredentials.publishingUserName;
         const password = publishCredentials.publishingPassword;
         const repo = `${this.site.enabledHostNames[1]}:443/${this.site.repositorySiteName}.git`;
         const remote = `https://${username}:${password}@${repo}`;
 
-        if (!workspace.rootPath) {
-            let input = await window.showErrorMessage(`You have not yet opened a folder to deploy.`);
-            return;
-        }
+
         let git = require('simple-git/promise')(workspace.rootPath);
 
         try {
             await git.init();
+            let status = await git.status();
+            if (status.files.length > 0) {
+                let input = await window.showInformationMessage(`There are uncommitted changes in local repo "${workspace.rootPath}". Commit?`, 'Yes');
+                if (input) {
+                    await git.add('./*');
+                    await git.commit('Deployed through VS Code');
```

Comment:
Is there a way to show the source control UI of VSC so users can have more options for commit (like customize the commit message, double checking the changes, etc.)?